sfs/util.py

```diff
@@ -491,3 +496,26 @@ def _count_walls_1d(a):
         wall_count = wall_count[N_mask, :]
 
     return xs, wall_count
+
+
+def spherical_hn2(n, z):
+    r"""Spherical Hankel function of 2nd kind.
+
+    Defined as http://dlmf.nist.gov/10.47.E6,
+
+    .. math:: \hankel{2}{n}{z} = \sqrt{\frac{\pi}{2z}} \Hankel{2}{n + 1/2}{z},
+
+    where :math:`\Hankel{2}{n}{}` is the Hankel function of the second kind.
+
+    Parameters
+    ----------
+    n : int, array_like
```

Comment:
IIRC, giving the datatype here is not what we normally do in the SFS toolbox.

We would normally just use `array_like` or we prepend the dimensions.

I know this is how SciPy does it, but shall we do it, too?

In NumPy I didn't find this style, I only found e.g. "int or 1-D array_like".

Reply:
The input can be a single int or a array_like thing including only ints.
For cases in which the input is a single value we provide `int` or `float` as description.
Regarding the dimensions, I guess they don't matter, but I'm not sure I just copied the description from SciPy.

By the way in the Toolbox I found also this one: [`int(N, 3) array_like`](https://github.com/sfstoolbox/sfs-python/blob/master/sfs/mono/drivingfunction.py#L404).

Reply:
Well I guess `int(N, 3) array_like` is a mistake (that was copy-pasted a few times), it doesn't make any sense at all.

According to the NumPy style we could write "int or array_like", but this wouldn't really make sense either, because an "int" is already a valid "array_like" (it gets converted to a zero-dimensional array).
We could write "array_like of int" or "array_like of ints" but I think this doesn't look quite nice.
We could write something like "array_like, dtype=int", which would be similar to how scikit learn does it, but I think that's more complicated than necessary.

I think it would be best to just write "array_like". It should be clear that the "order" has to be an integer.

Reply:
OK, I switched to `array_like`.